sig/matrix_gen.rbs

```diff
@@ -156,15 +156,15 @@ class Matrix
   def row_vectors : -> (Array[(Matrix | Vector)?])
   def column_vectors : -> (Array[(Matrix | Vector)?])
   def to_matrix : -> Matrix
-  def to_a : -> (Array[(Array[Integer] | Array[Integer] | Array[untyped] | String)?])
+  def to_a : -> (Array[(Array[Integer] | Array[Integer] | Array[untyped] | Integer | String)?])
```

Comment:
@mame: 
In this PR, you get the generated rbs file from `typeprof` after I've replaced `each_with_index` with `each.with_index`.
First commit shows result without any hint. This commit I tell it `Matrix@rows` and `Vector@elements` which should help a lot... but it makes some things *worse*. The perfect result here should be `Array[Array[numeric]]` (if it could reuse my type definition), or `Array[Array[Integer | Float | Rational | Complex]]` otherwise.

`Array[nil]` or `Array[String]` are very wrong...
Note that `Array[Integer] | Array[Integer]` is redundant.